client/src/components/Overview.jsx

```diff
@@ -22,19 +28,52 @@ export default class Overview extends React.Component {
     .catch(err => console.log(`unable to retrieve product info for product with id ${id}`, err));
   }
 
+  getProductStyles(id) {
+    axios.get(`/products/${id}/styles`)
+    .then(response => {
+      let currentStyle = response.data.results.find(style => style['default?'] === true);
+      this.setState({ product_styles: response.data.results, currentStyle });
+    })
+    .catch(err => console.log(`unable to retrieve product styles for product with id ${id}`, err));
+  }
+
+  getProductRating(id) {
```

Comment:
this getProductRating function is more elegant than mine. I also have to use this rating to show the stars. I'm thinking after we all finish our work. We can optimize later to use only one function and import it maybe to realize the same feature